### DISCUSSION TITLE
How to call a 'server' function w/o a form?

### DISCUSSION THREAD

**Discussion by aloukissas** [remix-run/remix#1321] [Q&A]
Many use cases here, but I'll give an example of what I'm trying to prototype. I'm integrating the [plaid link](https://github.com/plaid/react-plaid-link) library and I need to call a server function in a callback (see example project in plain reactjs [here](https://github.com/plaid/pattern/blob/4a26051cf9e286e62082c1e5a154d77026e33dd7/client/src/components/LaunchLink.tsx#L36)). In e.g. NextJS, this would translate in an "API route" call. I may have missed it in Remix's docs, but it looks the only way to trigger server functions is with submitting a form.

**Comment by c43721** [comment]
You can create a resource route (https://remix.run/docs/en/v1/guides/resource-routes#creating-resource-routes) and call it with fetch. This would be the 1:1 mapping from Next's API routes.

Basically, to create an "API route", in a new file export a loader and/or an action, **without** a default component export. That route will be treated as a resource route.

**Reply by sergiodxa** [reply]
Instead of using fetch you can also use the useFetcher hook, it gives you a load and submit method (for get and post requests) and it will handle the loading state for you.

**Reply by c43721** [reply]
Ah yes, the useFetcher hook 🙂- probably better to use this!

**Reply by aloukissas** [reply]
Thanks! Pretty complicated way to get around this (compared to pretty much everything else I've used) but if it works it works :)
